### PR TITLE
Spark edit

### PR DESCRIPTION
**Pull request by ttedeschi**

### CODE REVIEW

**Comment by dciangot** [comment]
I changed the master recently... please rebase on master branch to cure the conflicts

File: README.md [hunk 1]
```diff
@@ -12,4 +12,6 @@ helm repo update
 - [cachingondemand](stable/cachingondemand/README.md)
     - `helm install dodas/cachingondemand -v <myvalues>.yaml`
 - [htcondor](stable/htcondor/README.md)
-    - `helm install dodas/htcondor -v <myvalues>.yaml`
\ No newline at end of file
+    - `helm install dodas/htcondor -v <myvalues>.yaml`
+- [spark](stable/spark/README.md)
+    - `helm install dodas/spark -v <myvalues>.yaml`
```
Comment (dciangot) [inline README.md]: -f instead of -v

File: stable/spark/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+name: spark
+version: 1.0.0
+appVersion: 1.5.1
```
Comment (dciangot) [inline stable/spark/Chart.yaml]: put the correct version 2.4.4

File: stable/spark/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+description: Fast and general-purpose cluster computing system.
+home: http://spark.apache.org
+icon: http://spark.apache.org/images/spark-logo-trademark.png
+sources:
+  - https://github.com/kubernetes/kubernetes/tree/master/examples/spark
+  - https://github.com/apache/spark
+maintainers:
+  - name: lachie83
```
Comment (dciangot) [inline stable/spark/Chart.yaml]: put you name and email

File: stable/spark/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Apache Spark Helm Chart
+
+Apache Spark is a fast and general-purpose cluster computing system.
+
+* http://spark.apache.org/
+
+## Chart Details
+This chart will do the following:
+
+* 1 x Spark Master with port 8080 exposed with a nodePort service (webUi)
```
Comment (dciangot) [inline stable/spark/README.md]: nodeport cannot be less than 30000 usually, are you sure that you are exposing on 8080?

File: stable/spark/templates/spark-master-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,134 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: default-pyspark
+  namespace: default
```
Comment (dciangot) [inline stable/spark/templates/spark-master-deployment.yaml]: put as a value from values.yaml

Comment (dciangot) [reply stable/spark/templates/spark-master-deployment.yaml]: everywhere, not only here

File: stable/spark/templates/spark-master-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+spec:
+  type: {{ .Values.master.serviceType }}
+  ports:
+    - port: {{ .Values.master.servicePort }}
+      name: master
+      targetPort: {{ .Values.master.containerPort }}
+    - port: 8888
```
Comment (dciangot) [inline stable/spark/templates/spark-master-deployment.yaml]: pick jupyter ports from values.yaml

File: stable/spark/templates/spark-master-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+spec:
+  type: {{ .Values.webUi.serviceType }}
+  ports:
+    - port: {{ .Values.webUi.servicePort }}
+      nodePort: {{ .Values.webUi.nodePort }}
+      targetPort: {{ .Values.webUi.containerPort }}
+  externalIPs:
```
Comment (dciangot) [inline stable/spark/templates/spark-master-deployment.yaml]: add externalIPs only if specified in values. Take a look at [here](https://github.com/DODAS-TS/helm_charts/blob/master/stable/cachingondemand/templates/cache-redir-svc.yml#L26-L28) for reference

File: stable/spark/templates/spark-master-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+          command: ["start.sh"]
+          args: ["/usr/local/spark/bin/pyspark"]
+          env:
+            - name: PYSPARK_PYTHON
+              value: python3
+            - name: PYSPARK_DRIVER_PYTHON
+              value: ipython
+            - name: PYSPARK_DRIVER_PYTHON_OPTS
+              value: "notebook --NotebookApp.token={{ .Values.master.jupyter.token }}"
+          image: "cloudpg/spark-notebook:k8s-py3.6"
```
Comment (dciangot) [inline stable/spark/templates/spark-master-deployment.yaml]: let user specify image and tag. Look at [here](https://github.com/DODAS-TS/helm_charts/blob/master/stable/cachingondemand/templates/cache-server-pod.yml#L88) for reference

File: stable/spark/values.yaml [hunk 9]
```diff
@@ -3,10 +3,24 @@
 # Declare variables to be passed into your templates.
 
 
+
+#ClusterRoleBinding parameters
+clusterRoleBinding:
+    name: default-pyspark
+    namespace: default
```
Comment (dciangot) [inline stable/spark/values.yaml]: ok, but if you specify the namespace here, you have to set the same namespace in every other resources (pod, svc etc)

Comment (dciangot) [reply stable/spark/values.yaml]: @ttedeschi did you solved this?

Comment (ttedeschi) [reply stable/spark/values.yaml]: Solved, I removed the namespace line

Comment (dciangot) [reply stable/spark/values.yaml]: Mmm, @ttedeschi I meant to put namespace everywhere else :) not to remove it, sorry.

Comment (ttedeschi) [reply stable/spark/values.yaml]: @dciangot done!

**Pull request merged**